system_metrics_collector/test/topic_statistics_collector/test_subscriber_topic_statistics.cpp

```diff
@@ -73,14 +70,54 @@ class TestSubscriberTopicStatisticsNode
   }
 
   /**
-   * Return the container holding topic statistics collector.
+   * Check if individual topic statistics collectors are started.
    *
-   * @return data streucture containing the collectors
+   * @return true if all the collectoers are started, false otherwise
    */
-  std::vector<std::shared_ptr<topic_statistics_collector::TopicStatisticsCollector<
-      sensor_msgs::msg::Imu>>> GetCollectors() const
+  bool AreCollectorsStarted() const
   {
-    return statistics_collectors_;
+    for (const auto & collector : statistics_collectors_) {
+      if (!collector->IsStarted()) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Return number of collectors owned by this node.
+   *
+   * @return number of collectors now_nanoseconds*/
+  int GetCollectorCount() const
+  {
+    return statistics_collectors_.size();
+  }
+
+  /**
+   * Return the statistic data collected by all individual collectors in this node.
+   *
+   * @return statistic data collected by all collectors
+   */
+  std::vector<moving_average_statistics::StatisticData> GetCollectorData() const
+  {
+    std::vector<moving_average_statistics::StatisticData> data;
+    for (const auto & collector : statistics_collectors_) {
+      data.push_back(collector->GetStatisticsResults());
+    }
+    return data;
+  }
+
+  void SpinUntilMessageReceived()
```

Comment:
This works. Nit: I'd prefer using the PromiseSetter pattern used elsewhere but only for consistency.

Reply:
Nevermind this is great, went over in-person :-)